### PR TITLE
AF: higher LR sweep (7e-4, 8e-4) on vol-10x+EMA champion

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The AF champion uses lr=6e-4 (AdamW). Previous higher-LR tests (lr=8e-4 → +110% surface, lr=1e-3 → catastrophic) were run WITHOUT EMA and WITHOUT vol-weight=10x. EMA=0.999 smooths the parameter trajectory and gc=1.0 clips gradient spikes, which together may **expand the viable LR range upward**. A modestly higher LR could improve convergence speed and find deeper loss basins — particularly for the volume channel which has 10x loss amplification and thus experiences effectively lr=7e-3/8e-3 on its gradient signal.

Your own lower-LR sweep (#3234) showed that lr<6e-4 causes underfitting (surface +38-66%, vol +76-111%). This brackets the optimum from below. We now need to bracket from above under the current EMA+vol-10x+gc=1.0 regime to determine whether 6e-4 is actually optimal or whether the pre-EMA dead ends at 8e-4 no longer apply.

## Instructions

**Smoke test first (3 epochs)** — verify loss finite, gradients stable.

**Trial 1 — lr=7e-4:**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 7e-4 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --wandb_group af-higher-lr-vol10x
```

**Trial 2 — lr=8e-4:**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 8e-4 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --wandb_group af-higher-lr-vol10x
```

Report `val_primary/surface_mse` AND `full_val/volume_mse` for both trials. Target: vol_mse < **0.0017** without regressing surface_mse above **0.000296**.

## Baseline

Current best AirfRANS metrics (PR #3135):

| Metric | Value | Epoch |
|--------|-------|-------|
| val_primary/surface_mse | **0.000296** | 704 |
| val_primary/vol_mse | **0.002039** | 743 |

- W&B run: `sh2zyfwr`
- Config: 2L/256d/4H, AdamW lr=6e-4, T_max=50, gc=1.0, WD=1e-2, EMA=0.999, vol-weight=10x, Fourier

Reproduce:
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --vol-loss-weight 10.0
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3268 because ADVISOR: Closing per issue #3283 directive: 'Stop higher-LR arms' for AirfRANS. lr=6e-4 is bracketed. Reassigning to priority lane..

**Comment by tcapelle** [comment]
STUDENT jin:

## Results

### Trial 1 — lr=7e-4

| Metric | Baseline | Result | Delta |
|--------|----------|--------|-------|
| val_primary/surface_mse | **0.000296** | 0.001109 | **+274%** |
| full_val/volume_mse | **0.002039** | 0.007189 | **+253%** |

- **Best epoch:** 229 (out of 769 trained — **diverged to NaN** after ep229)
- **W&B run:** [`0ux26l2z`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/0ux26l2z)
- **Grad norms:** Mean ~65 at best epoch (all 360 batches clipped), escalated to NaN by late training
- **Command:**
  ```bash
  cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
    --dataset airfrans --airfrans-task full \
    --optimizer adamw --lr 7e-4 \
    --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
    --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
    --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 \
    --agent jin --wandb-name "jin/af-lr7e4-vol10x-ema" --wandb-group af-higher-lr-vol10x
  ```

### Trial 2 — lr=8e-4

| Metric | Baseline | Result | Delta |
|--------|----------|--------|-------|
| val_primary/surface_mse | **0.000296** | 0.000857 | **+189%** |
| full_val/volume_mse | **0.002039** | 0.002964 | **+45%** |

- **Best epoch:** 638 (out of 766 trained — stable throughout, no divergence)
- **W&B run:** [`zqrl1dr5`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/zqrl1dr5)
- **Grad norms:** Mean ~12.8 at best epoch (336/360 batches clipped), well-controlled
- **Command:**
  ```bash
  cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py \
    --dataset airfrans --airfrans-task full \
    --optimizer adamw --lr 8e-4 \
    --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
    --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
    --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 \
    --agent jin --wandb-name "jin/af-lr8e4-vol10x-ema" --wandb-group af-higher-lr-vol10x
  ```

### What happened

**Both trials are clear negatives.** Neither LR came close to the baseline targets (surface < 0.000296, vol < 0.0017).

Counterintuitively, lr=8e-4 **outperformed** lr=7e-4 on both metrics (surface 0.000857 vs 0.001109, vol 0.002964 vs 0.007189), and lr=7e-4 diverged to NaN while lr=8e-4 stayed stable. This likely reflects seed-dependent basin structure rather than a systematic lr=8e-4 advantage — the pre-EMA result (lr=8e-4 → +110% surface) also showed instability.

Key observations:
- **lr=7e-4 diverged catastrophically** after epoch 229 with grad_norm_mean escalating to NaN. EMA + gc=1.0 was NOT sufficient to stabilize it. The mean grad norm at best epoch was ~65 (vs ~12.8 for lr=8e-4), suggesting lr=7e-4 sits in a particularly rough loss landscape region.
- **lr=8e-4 was stable but found a shallower basin.** It ran all 766 epochs without divergence but plateaued at 2.9x the baseline surface MSE. The vol-weight=10x amplification means the effective LR on volume gradients is 8e-3, which may be too aggressive.
- **lr=6e-4 appears to be the optimum** for the EMA+vol-10x+gc=1.0 regime. The lower LR sweep (#3234) showed underfitting below 6e-4, and this sweep shows degradation above 6e-4. The viable LR window is narrow — the champion sits at or very near the peak.

### Suggested follow-ups
- **Finer grid around 6e-4:** lr=6.5e-4 could test whether there's headroom in the narrow zone between 6e-4 (champion) and 7e-4 (diverged). However, given the magnitude of degradation at 7e-4, the expected gain is small.
- **Warmup + higher LR:** A linear warmup period (e.g., 10-20 epochs) before reaching 7e-4 could allow the model to find a stable basin first, potentially preventing the early divergence seen in Trial 1.
- **gc=0.5 + higher LR:** Tighter gradient clipping might enable higher LRs by preventing the gradient spikes that destabilized lr=7e-4.
